benchmark/david_benchmark.py

```diff
@@ -0,0 +1,151 @@
+# © Crown Copyright GCHQ
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Benchmark performance of different coreset algorithms on pixel data from an image.
+
+The benchmarking process follows these steps:
+1. Load an input image and downsample it using a specified factor.
+2. Convert to grayscale and extract non-zero pixel locations and values.
```

Comment:
What is meant by 'non-zero pixel locations and values' ? Does this mean pixels that have value above zero? i.e., are not completely black?

Reply:
This is a typo, removed `non-zero`.